libs/error-stack/src/fmt.rs

```diff
@@ -133,6 +133,148 @@
+//! `6`, `1` has more than 1 child.
+//!
+//! A group is a list of nodes where each node shares a common immediate context parent that has
+//! more than `1` child, this means that `(2, 6)` is a group (they share `1` as an immediate context
+//! parent), while `(3, 4, 6)` is not. `(3, 4, 6)` share the same parent with more than 1 child
+//! (`1`), but `1` is not the immediate context parent of `3` and `4` (`2`) is. In the more detailed
+//! example `(Dᶜ, Hᶜ, Cᶜ)` is considered a group because they share the same *immediate* context
+//! parent `Aᶜ`, important to note is that we only refer to immediate context parents, `Fᵃ` is the
+//! immediate parent of `Iᶜ`, but is not a [`Context`], therefore to find the immediate context
+//! parent, we travel up the tree until we encounter our first [`Context`] node. Groups always
+//! contain lists, for the sake of clarity this explanation only shows the first element.
+//!
+//! Additional rules that apply:
+//! * lists are never empty
+//! * lists are nested in groups
+//! * groups are always preceded by lists
+//! * groups are ordered left to right
+//!
+//! Using the aforementioned delimiters for lists and groups the end result would be:
+//!
+//! Overview Tree: `[0, 1] ([2] ([3], [4, 5]), [6, 7, 8])`
+//! Detailed Tree: `[Aᶜ] ([Dᶜ], [Hᶜ], [Iᶜ])`
+//!
+//! Attachments are not ordered by insertion order but by depth in the tree. Each context uses the
+//! attachments that are it's parents until the next context node. If attachments are shared between
+//! multiple contexts, they are duplicated and output twice.
+//!
+//! ### Output Formatting
+//!
+//! Lists are guaranteed to be non-empty and have at least a single context. The context is the
+//! heading of the whole list, while all other contexts are intended. The last entry in that
+//! indentation is (if present) the group that follows, taking the detailed example this means that
+//! the following output would be rendered:
+//!
+//! ```text
+//! Aᶜ
+//! │
+//! ╰┬▶ Dᶜ
+//!  │  ├╴Bᵃ
+//!  │  ╰╴Cᵃ
+//!  │
+//!  ├▶ Hᶜ
+//!  │  ├╴Bᵃ
+//!  │  ├╴Eᵃ
+//!  │  ├╴Fᵃ
+//!  │  ╰╴Gᵃ
+//!  │
+//!  ╰▶ Iᶜ
+//!     ├╴Bᵃ
+//!     ├╴Eᵃ
+//!     ╰╴Fᵃ
+//! ```
+//!
+//! Groups are visually represented as an additional distinct indentation for other contexts in the
+//! preceding list, taking the overview tree this means:
+//!
+//! ```text
+//! 0
+//! ├╴Attachment
+//! │
+//! ├─▶ 1
+//! │   ╰╴Attachment
+//! │
+//! ╰┬▶ 2
+//!  │  │
+//!  │  ╰┬▶ 3
+//!  │   │
+//!  │   ╰▶ 4
+//!  │      │
+//!  │      ╰─▶ 5
+//!  ╰▶ 6
+//!     │
+//!     ├─▶ 7
+//!     │
+//!     ╰─▶ 8
+//! ```
+//!
+//! Attachments have been added to various places to simulate a real usecase with attachment and
+//! visualise their placement.
+//!
+//! The spacing and characters used are chosen carefully, as even when you have a context vs. group
+//! of contexts (as seen above) alignment is still ensured.
```

Comment:
I think this is not the case. `1` is the parent of the `(2, 3)` group but they are rendered in the same column.